### PR TITLE
Sub getProwJobSpec return types are named

### PR DESCRIPTION
**Pull request by chaodaiG**
Follow up #27581

/cc @listx

### CODE REVIEW

File: prow/pubsub/subscriber/subscriber.go [hunk 1]
```diff
@@ -150,13 +150,13 @@ func (m *pubSubMessage) nack() {
 
 // jobHandler handles job type specific logic
 type jobHandler interface {
-	getProwJobSpec(cfg prowCfgClient, pc *config.InRepoConfigCacheHandler, pe ProwJobEvent) (*v1.ProwJobSpec, map[string]string, map[string]string, error)
+	getProwJobSpec(cfg prowCfgClient, pc *config.InRepoConfigCacheHandler, pe ProwJobEvent) (jobSpec *v1.ProwJobSpec, labels map[string]string, annotations map[string]string, err error)
```
Comment (listx) [inline prow/pubsub/subscriber/subscriber.go:153]: Nit: I was thinking more along the lines of https://yourbasic.org/golang/type-alias/. This is still better than the existing situation though, so +1.

**Review by listx — COMMENTED** [review]
/lgtm
/hold

File: prow/pubsub/subscriber/subscriber.go (hunk 1 above)
Comment (chaodaiG) [reply prow/pubsub/subscriber/subscriber.go:153]: ack. The type alias change would be more involved imo, we can consider that in the future

**Pull request merged**